src/applications/burials-ez/config/chapters/05-additional-information/deathCertificate.js

```diff
@@ -15,18 +15,17 @@ export default {
     deathCertificate: {
       ...burialUploadUI('Upload the Veteran’s death certificate'),
       'ui:required': form => {
-        const option1 = form?.burialAllowanceRequested?.service === true;
-        const option2 = form?.locationOfDeath?.location !== 'vaMedicalCenter';
-        const option3 = Boolean(
-          form?.burialAllowanceRequested?.service === true &&
-          form?.locationOfDeath?.location === 'vaMedicalCenter'
-            ? 0
-            : 1,
+        const isClaimingBurialAllowance =
+          form['view:claimedBenefits'].burialAllowance;
+        const serviceRequested =
+          form.burialAllowanceRequested?.service === true;
+        const locationIsVaMedicalCenter =
+          form.locationOfDeath?.location === 'vaMedicalCenter';
+        return !(
+          isClaimingBurialAllowance &&
+          serviceRequested &&
+          locationIsVaMedicalCenter
```

Comment:
@Midge-dev Wanted to get your opinion on this code change. I added an additional check for step 4 > Costs you paid for > Burial and funeral costs option checked.

Otherwise, a user could initially select the option, then selected service related, then go back uncheck the burial allowance option, and proceed with another option instead.

Rare edge case. I think this covers it, but wanted to get your take too.

Reply:
Good call out. I didn't account for this scenario at the time. LGTM